packages/flutter/lib/src/material/button.dart

```diff
@@ -421,3 +445,34 @@ class MaterialButton extends StatelessWidget {
     properties.add(new FlagProperty('enabled', value: enabled, ifFalse: 'disabled'));
   }
 }
+
+/// Redirects the position passed to [RenderBox.hitTest] to the center of the widget.
+///
+/// The primary purpose of this widget is to allow padding around [Material] widgets
+/// to trigger the child ink feature without increasing the size of the material.
+class _RedirectingHitDetectionWidget extends SingleChildRenderObjectWidget {
+  const _RedirectingHitDetectionWidget({this.constraints, Widget child}) : super(child: child);
+
+  final BoxConstraints constraints;
+
+  @override
+  RenderObject createRenderObject(BuildContext context) {
+    return _RedirectingHitDetectionRenderObject(constraints);
+  }
+
+  @override
+  void updateRenderObject(BuildContext context, covariant _RedirectingHitDetectionRenderObject renderObject) {
+    renderObject.additionalConstraints = constraints;
+  }
+}
+
+class _RedirectingHitDetectionRenderObject extends RenderConstrainedBox {
```

Comment:
convention for naming these is "RenderFoo", with the widget being called "Foo".

Reply:
Fixed